version: '3.8'
services:
  steamate:
    container_name: steamate
    build:
      context: ./steamate
      dockerfile: Dockerfile.wsgi
      args:
        PYTHON_VERSION: 3.12.9
      cache_from:
        - steamate:latest
    volumes:
      - ./steamate:/app
    env_file:
      - .env
    ports:
      - "8000:8000"
    networks:
      - steamate-network
  asgi-server:
    build:
      context: .
      dockerfile: steamate/Dockerfile.asgi
    ports:
      - "8001:8001"
    networks:
      - steamate-network
    depends_on:
      - redis

  redis:
    image: redis:alpine
    container_name: redis
    ports:
      - "6379:6379"
    networks:
      - steamate-network

  celery:
    build:
      context: ./steamate
      dockerfile: Dockerfile.wsgi
    command: ["celery", "-A", "config", "worker", "--loglevel=info", "--concurrency=4"]
    volumes:
      - ./steamate:/app
    working_dir: /app
    environment:
      - PYTHONPATH=/app
    env_file:
      - .env
    depends_on:
      - steamate
      - redis
    networks:
      - steamate-network

  nginx:
    build:
      context: ./nginx
      args:
        NGINX_VERSION: 1.24.0
    container_name: nginx
    restart: "on-failure"
    ports:
      - "80:80"
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf
    depends_on:
      - steamate
    networks:
      - steamate-network
    

networks:
  steamate-network:
    driver: bridge
